pennylane/templates/state_preparations/basis_qutrit.py

```diff
@@ -112,7 +113,10 @@ def compute_decomposition(basis_state, wires):  # pylint: disable=arguments-diff
         """
 
         op_list = []
+        tshift = qml.math.array([[0, 0, 1], [1, 0, 0], [0, 1, 0]])
```

Comment:
Does this scale worse or better than the default case? If worse, then should we limit this only to `jit` case (and use the default behaviour otherwise)?

Reply:
Good suggestion. I'm changing it to default to the old decomposition, and only decompose to `qml.QutritUnitary` when jit is enabled.